src/main/scala/com/gu/contentapi/services/PodcastLookup.scala

```diff
@@ -46,13 +51,25 @@ object PageLookup extends StrictLogging {
       Await.result(
         makeCapiQuery(filePath) map {
           case SuccessfulQuery(sr) => {
-            sr.results.headOption foreach { content => cache.put(filePath, content.webUrl) }
-            sr.results.headOption map (_.webUrl)
+            for {
+              webUrl <- sr.results.headOption map (_.webUrl)
+              tags <- sr.results.headOption.map(_.tags)
+              podcastName <- extractPodcastSeriesName(tags)
+            } yield {
+              // update cache and return info
+              sr.results.headOption foreach { content => cache.put(filePath, PodcastInfo(webUrl, podcastName)) }
+              PodcastInfo(webUrl, podcastName)
+            }
           }
-          case FailedQuery(err) => Some(err)
+          case FailedQuery(err) => None
         }, 2.seconds
       )
     }
   }
 
+  def extractPodcastSeriesName(tags: Seq[Tag]): Option[String] = {
+
+    tags.foldLeft[Option[String]](None) { (_, tag) => if (tag.podcast.isDefined) return Some(tag.webTitle) else None }
```

Comment:
The use of foldLeft here makes the code more obscure! I'm sure this can be written more simply.